service/lib/agama/storage/manager.rb

```diff
@@ -49,148 +46,87 @@ module Storage
     class Manager
       include WithLocale
       include WithIssues
-      include WithProgress
-      include Yast::I18n
+      include WithProgressManager
 
       # @return [Agama::Config]
       attr_reader :product_config
 
       # @return [Bootloader]
       attr_reader :bootloader
 
-      # Constructor
-      #
       # @param product_config [Agama::Config]
       # @param logger [Logger, nil]
       def initialize(product_config, logger: nil)
-        textdomain "agama"
-
         @product_config = product_config
         @logger = logger || Logger.new($stdout)
         @bootloader = Bootloader.new(logger)
-
-        register_progress_callbacks
       end
 
-      # Whether the system is in a deprecated status
-      #
-      # The system is usually set as deprecated as effect of managing some kind of devices, for
-      # example, when iSCSI sessions are created.
-      #
-      # A deprecated system means that the probed system could not match with the current system.
-      #
-      # @return [Boolean]
-      def deprecated_system?
-        !!@deprecated_system
+      def activated?
+        !!@activated
       end
 
-      # Sets whether the system is deprecated
-      #
-      # If the deprecated status changes, then callbacks are executed and the issues are
-      # recalculated, see {#on_deprecated_system_change}.
-      #
-      # @param value [Boolean]
-      def deprecated_system=(value)
-        return if deprecated_system? == value
+      def reset_activation
+        Y2Storage::Luks.reset_activation_infos
+        @activated = false
```

Comment:
I'm not sure whether setting `activated` to false is correct. Resetting the LUKS activation info gives a second chance to introduce LUKS passwords, but it does not invalidate any previous activation.